src/frontend/src/pages/CreateWorkPackagePage/CreateWPForm.tsx

```diff
@@ -3,133 +3,39 @@
-  wbsNum: Dispatch<SetStateAction<string>>;
-  crId: Dispatch<SetStateAction<string>>;
-  startDate: Dispatch<SetStateAction<Date>>;
-  duration: Dispatch<SetStateAction<number>>;
-}
 
 const CreateWPForm: React.FC = () => {
   const history = useHistory();
   const auth = useAuth();
-  const query = useQuery();
 
-  const [name, setName] = useState(query.get('name') ?? '');
-  const [projectWbsNum, setWbsNum] = useState(query.get('wbs') ?? '');
-  const [crId, setCrId] = useState(query.get('crId') || '');
-  const [startDate, setStartDate] = useState<Date>(new Date());
-  const [duration, setDuration] = useState(numberParamPipe(query.get('duration')) ?? -1);
-  const [dependencies, setDependencies] = useState<string[]>([]);
-  const [expectedActivities, setExpectedActivities] = useState<string[]>([]);
-  const [deliverables, setDeliverables] = useState<string[]>([]);
   const { isLoading, mutateAsync } = useCreateSingleWorkPackage();
 
   if (isLoading) return <LoadingIndicator />;
 
-  const depUtils: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = dependencies.slice();
-      clone.push(val);
-      setDependencies(clone);
-    },
-    remove: (idx) => {
-      const clone = dependencies.slice();
-      clone.splice(idx, 1);
-      setDependencies(clone);
-    },
-    update: (idx, val) => {
-      const clone = dependencies.slice();
-      clone[idx] = val;
-      setDependencies(clone);
-    }
-  };
-
-  const eaUtils: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = expectedActivities.slice();
-      clone.push(val);
-      setExpectedActivities(clone);
-    },
-    remove: (idx) => {
-      const clone = expectedActivities.slice();
-      clone.splice(idx, 1);
-      setExpectedActivities(clone);
-    },
-    update: (idx, val) => {
-      const clone = expectedActivities.slice();
-      clone[idx] = val;
-      setExpectedActivities(clone);
-    }
-  };
-
-  const delUtils: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = deliverables.slice();
-      clone.push(val);
-      setDeliverables(clone);
-    },
-    remove: (idx) => {
-      const clone = deliverables.slice();
-      clone.splice(idx, 1);
-      setDeliverables(clone);
-    },
-    update: (idx, val) => {
-      const clone = deliverables.slice();
-      clone[idx] = val;
-      setDeliverables(clone);
-    }
-  };
-
-  const states = {
-    name: setName,
-    wbsNum: setWbsNum,
-    crId: setCrId,
-    startDate: setStartDate,
-    duration: setDuration
-  };
-
-  const initialValues = {
-    name,
-    wbsNum: projectWbsNum,
-    crId,
-    duration
-  };
-
-  const handleSubmit = async (e: any) => {
-    e.preventDefault();
+  const handleSubmit = async (data: any) => {
+    const { name, startDate, duration, crId, dependencies, wbsNum } = data;
+    const expectedActivities = data.expectedActivities.map((bullet: { bulletId: number; detail: string }) => bullet.detail);
```

Comment:
check out `form.ts`

Reply:
I didn't use `mapBulletsToPayload` in `form.ts` because it was causing crashes after submitting, I think this specific backend route expects the expectedActivities and deliverables to be a list of strings instead of an object with an id and detail.

Reply:
oh i see, missed that